source/endpoint-reference/endpoint-folders.rst

```diff
@@ -0,0 +1,191 @@
+only contain a flat list of all the existing dataset variables.
+
+As variables get grouped into folders the folders root will continue showing
+the ungrouped variables and existing folders.
+
+
+Subfolder endpoint
+~~~~~~~~~~~~~~~~~~
+
+``/datasets/{id}/folders/{folder_id}/``
+
+All subfolders' URLs follow straight from the folders' root URL regardless
+of their nested location in the folders' tree.
+
+All subfolders' payload has the same shape as the root catalog.
+
+Folder payload
+--------------
+
+All folders, trash and root return a similar Shoji catalog with the same shape
+which includes the following members:
+
+======== ======= ===============================================================
+Member   Type    Description
+======== ======= ===============================================================
+size     int     Indicates the number of total variables stored under this
+                 folder and its subfolders.
+-------- ------- ---------------------------------------------------------------
+body     object  Contains a `name` attribute, which is the folder's name.
+                 The root folder has an empty string as name.
+-------- ------- ---------------------------------------------------------------
+index    object  Behaves like a standard Shoji catalog containing the children
+                 of the folder keyed by URL. This catalog will contain a mixed
+                 set of folder and variables URLs. Clients should differentiate
+                 each of the objects by looking at the tuple's `type` attribute.
+                 Folders will have type "folder" while variables will contain
+                 their variables type.
+-------- ------- ---------------------------------------------------------------
+graph    list    A flat list containing the list of URLs of variables and
+                 subfolders on the given folder.
+======== ======= ===============================================================
+
+JSON example:
+
+.. language_specific::
+   --JSON
+   .. code:: json
+
+      {
+          "element": "shoji:catalog",
+          "self": "https://app.crunch.io/api/datasets/5ee0a0/variables/",
+          "size": 10000,
+          "body": {
+              "name": "Folder name"
+          },
+          "index": {
+              "https://app.crunch.io/datasets/abcdef/variables/123/": {
+                  "name": "Birth Year",
+                  "derived": false,
+                  "discarded": false,
+                  "alias": "birthyear",
+                  "type": "numeric",
+                  "id": "123",
+                  "notes": "",
+                  "description": "In what year were you born?"
+              },
+              "https://app.crunch.io/datasets/abcdef/folders/qwe/": {
+                  "name": "Subfolder name"
+              },
+              "https://app.crunch.io/datasets/abcdef/variables/456/": {
+                  "subvariables_catalog": "aad4ad/subvariables/",
```

Comment:
This doesn't make sense to me. First, it looks like you need to pick 456 or aad4ad as the variable id; second, this URL can't be relative to the URL on the line above it..it has to be relative to the "self" URL.

Reply:
Fixing